lib/hijack/wrap_observers.js

```diff
@@ -1,13 +1,30 @@
 wrapOplogObserveDriver = function(proto) {
-  // Track the polled documents. This is reflect to the RAM size and 
+  // Track the polled documents. This is reflect to the RAM size and
   // for the CPU usage directly
   var originalPublishNewResults = proto._publishNewResults;
   proto._publishNewResults = function(newResults, newBuffer) {
+    var data = [];
+    newResults.forEach(function(item){
```

Comment:
Don't loop this. Find some way to send these two values to the `getSize()`  at once. Like:

~~~js
var docSize = Kadira.docSzCache.getSize(coll, query, opts, newResults, newBuffer);
~~~

And modify getSize to work with IDmaps too.

With that, we can avoid some loops.